### PR TITLE
Fix Screenshots in macOS Safari - adapt to rounded corners

### PR DESCRIPTION
**Pull request by htho**
## Purpose
#8154 addresses the issue, where screenshots are not cropped on macOS (14) Safari.
The problem is, that the screenshot mark can not be found.
This is because part of the mark is not visible because of the rounded corners in macOS.

312752882-4a8a5548-85e3-4805-855e-608366c74c8b

## Approach
The solution is to increase `MARK_RIGHT_MARGIN` in `src/screenshots/constants.js`.
I found that `25` is the lowest value that works for all resolutions I was able to test.

* In order for the tests to work I parametrized the tests to depend on `MARK_RIGHT_MARGIN`.
* ~~I enabled the screenshot-tests that were skipped in the safari browser~~
  * ~~fix tests~~
  * ~~retina aware~~
  * ~~color-profile aware~~

## References
#8154 

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [x] Make sure that existing tests do not fail

### CODE REVIEW

**Comment by htho** [comment]
Regarding Tests: functional tests seem unstable on my machine, therefore I hope to get better results on the functional test from TestCafes Test Actions on GitHub.

File: test/functional/assertion-helper.js [hunk 1]
```diff
@@ -59,10 +60,18 @@ function checkScreenshotFileCropped (filePath) {
...
-                   hasPixel(png, RED_PIXEL, 0, height - 1) && hasPixel(png, RED_PIXEL, 49, height - 50) && hasPixel(png, GREEN_PIXEL, 50, height - 51);
+            // NOTE: tests do not work on retina displays since test code is not aware of this
+            // NOTE: tests check for absolute rgb values therefore the display needs to be set to an sRGB color profile (macOS)
+            return (
+                // topLeft
+                hasPixel(png, RED_PIXEL, 0, 0) && hasPixel(png, RED_PIXEL, 49, 49) && hasPixel(png, GREEN_PIXEL, 50, 50) ||
+                // topRight
+                hasPixel(png, RED_PIXEL, width - 1, 0) && hasPixel(png, RED_PIXEL, width - 50, 49) && hasPixel(png, GREEN_PIXEL, width - 51, 50) ||
+                // bottomRightRounded
+                hasPixel(png, RED_PIXEL, width - 1 - MARK_RIGHT_MARGIN, height - 1) && hasPixel(png, RED_PIXEL, width - 1, height - 1 - MARK_RIGHT_MARGIN) && hasPixel(png, RED_PIXEL, width - 50, height - 50) && hasPixel(png, GREEN_PIXEL, width - 51, height - 51) ||
```
Comment (htho) [inline test/functional/assertion-helper.js:71]: this is where I changed the checks, not to look at the corner, but at the edges near the corner

**Comment by htho** [comment]
Hmm. Looking at the failed tests, the test machine seems to either also have/emulate a retina screen or to use a color profile other than sRGB.

The test looks at the corners of the screenshot, at coordinates inferred from the set size.
For retina screens the screenshot has a greater resolution than the size set. Therefore the coordinates don't match. The test looks pixels in the bottom right corner, but it actually looks somewhere in the center of the screenshot.
The top-left pixel should be correct, but somehow safari renders the first row of pixels in a darker red tone than set.

If the color profile is set to something else than sRGB then the RGB color values don't match - the red box is red, but not exactly the expected value of #FF0000.

@PavelMor25 can you check the specs of the macOS machine for me?

Also: I don't understand why the Docker tests fail.

**Comment by PavelMor25** [comment]
Hello @htho,
 
Thank you for your pull request. We will review it.
 
The `Test Functional (Local Safari)` workflow utilizes [GitHub-hosted runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners). In our case, tests run on macOS 13. This provides all the information about the specifications of the macOS machine:
 
- Image: macos-13
- Version: 20240327.2
- Included Software: https://github.com/actions/runner-images/blob/macos-13/20240327.2/images/macos/macos-13-Readme.md
- Image Release: https://github.com/actions/runner-images/releases/tag/macos-13%2F20240327.2
 
Regarding Docker tests, you need to merge the latest changes from the master branch. This should fix the tests.

**Comment by htho** [comment]
@PavelMor25 is it possible for your, to trigger the tests?

**Comment by htho** [comment]
@PavelMor25 I reset my branch to [the commit](https://github.com/htho/testcafe/commit/007b62e2942e23fabe0f32a802d3f6ed1da1c13d) that changes `MARK_RIGHT_MARGIN`.

Are you sure, you also want me to remove [the commit](https://github.com/htho/testcafe/commit/d43e1ab285fe60f9c32b0e0b0d8dc90a06996612) that parameterizes the server tests?
Because without this commit tests in `crop-test.js` fail-

**Review by PavelMor25 — APPROVED** [review]

**Comment by htho** [comment]
great thanks!

**Pull request merged**